### PR TITLE
Fix: Activity feed table fixes

### LINKED ISSUE #3487: [Dashboard] Dividers aren't shown on mobile in the Recent Activity section of the dashboard
## Steps to reproduce

1. Open the dashboard on a mobile device
Navigate to the Recent Activity section
Observe the absence of dividers between items

## Expected behaviour

* Dividers are shown on mobile in the Recent Activity section of the dashboard

## Actual behaviour

* Dividers aren't shown on mobile in the Recent Activity section of the dashboard

image

### LINKED ISSUE #3495: Dashboard pagination resets when requesting new data
## Steps to reproduce

1. Ensure you several pages of activity in your colony.
2. Navigate through the activity feed on the dashboard.
3. Note if you can get to the end without the pagination resetting back to page 1.

## Expected behaviour

* Activity feed pagination should allow you to navigate through all the pages.

## Actual behaviour

* As you navigate through the activity feed with a large list, the pagination resets.

[pagination-resets.webm](https://github.com/user-attachments/assets/0538c1b0-7d57-41f8-ab32-d41ba88a5493)

### LINKED ISSUE #3499: [Dashboard - Latest activity] Action menu isn’t fully opened on mobile when opened at the bottom
## Steps to reproduce

1. Go to the Activity (or dashboard - Recent activity) screen on a mobile device.
2. Scroll to an item near the bottom of the page.
3. Click to open the action menu for that item.
4. Scroll screen up and down.

## Expected behaviour

* The action menu should fully open and be fully visible on the screen.

## Actual behaviour

* The action menu is partially cut off when opened near the bottom of the screen, making it difficult to interact with.

image

https://github.com/user-attachments/assets/98d8c1c1-1413-4086-bd1c-2a796b3259b5

### LINKED ISSUE #3552: QA Tables: Closing rows between mobile and desktop viewport changes
Raising this with the caveat that it's an edge case and a seemingly low priority.

## Steps to reproduce

1. Navigate to the activity/ dashboard pages in the mobile viewport.
2. Open a row within the activity tables on either page.
3. In the browser, switch back to the desktop viewport size.
4. Note that the mobile row is still open in the table and displaying multiple meatball menus.
5. Note the overflow of table content outside of the table container.

## Expected behaviour

* When switching back to the desktop viewport, ideally the table would reset to the correct UI and no longer show the mobile content row expanded. The table content should not overflow and break the UI.

## Actual behaviour

* The mobile expanded row displays in the desktop table along with the desktop meatball icon. Table content overflows at the bottom and breaks the UI. 

image

image

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- This PR aims to solve multiple issues related to the activity table:
- [X] Dashboard pagination resets when requesting new data
- [X] Expanded rows not closing between mobile and desktop viewport changes
- [X] Row dividers not showing up on mobile or Safari
- [X] Actions menu isn’t fully opened on mobile

Please keep in mind I did resist the urge to refactor the `Table` component.

## Testing

TODO: Please test all these scenarios are resolved. For the first issues you can use `Chrome` as on this branch a bug regarding the pagination is still present (fixed by ), for the second issue any browser would do, but for the last 2 issues please use Safari or a mobile simulator - shout if you need help configuring a mobile simulator 😉 

#### Issue 1: Dashboard pagination resets when requesting new data

* Step 1.1. Let us test the pagination no longer resets when requesting new data. In `ColonyHome` please update the `pageSize` to `3` for the `RecentActivityTable` component - thus you'll have more pages of data to test this.
* Step 1.2. Click as fast as possible on the `Next` button to make sure the correct data loads and the pagination is not reset to `1`

https://github.com/user-attachments/assets/2c076ca1-16be-43a7-b769-9242267cf452

#### Issue 2: Expanded rows not closing between mobile and desktop viewport changes

* Step 2.1. Now lets make sure we have a mobile viewport.
* Step 2.2. Expand a few of the rows.
* Step 2.3. Now switch to a desktop view and make sure the expanded content is no longer visible.
* Step 2.4. You can go back to a mobile view and the rows should still be unexpanded

https://github.com/user-attachments/assets/32b5ec52-7924-4ab0-9366-8ecd067bac2d

#### Issue 3: Row dividers not showing up on mobile or Safari

* Step 3.1. Now let's go to `Safari` or a mobile simulator.
* Step 3.2. Check the row dividers are visible.

Screenshot 2024-10-31 at 17 12 09

#### Issue 4: Actions menu isn’t fully opened on mobile

* Step 4.1. Again let's use `Safari` or a mobile simulator.
* Step 4.2. Go to one row that is partially visible (preferably at the bottom of the screen)
* Step 4.3. Open up the actions menu and check all options are visible and not cut off.
* Step 4.4. Try opening more options at different table positions and check the menu is visible.
 

https://github.com/user-attachments/assets/8e4f4790-1d01-4a32-ac85-46bd16624995

## Diffs

**New stuff** ✨

* `TableRowDivider` component. Unfortunately this is needed because Safari and `table`, `tr`, `td` etc. elements don't respond well to `absolute` position. 

Resolves #3495
Resolves #3552
Resolves #3487
Resolves #3499

### CODE REVIEW

File: src/components/v5/common/Table/Table.tsx [hunk 1]
```diff
@@ -406,6 +418,12 @@ const Table = <T,>({
                           </td>
                         </tr>
                       )}
+                      {
+                        /** Unfortunately Safari is not yet that friendly to allow the usage of absolutely positioned (pseudo)-element inside tables
+                         * So, for the moment, this is our best shot for showing borders with paddings from the tr margins
+                         */
+                        withNarrowBorder && <TableRowDivider />
```
Comment (rumzledz) [inline src/components/v5/common/Table/Table.tsx:425]: 👍

File: src/components/common/ColonyActionsTable/hooks/useActionsTableProps.tsx [hunk 2]
```diff
@@ -83,6 +83,10 @@ export const useActionsTableProps = (
     original: { transactionHash },
   }) => ({
     disabled: loading,
+    dropdownPlacementProps: {
+      withAutoTopPlacement: true,
```
Comment (rumzledz) [inline src/components/common/ColonyActionsTable/hooks/useActionsTableProps.tsx:87]: ✨

**Review by rumzledz — APPROVED** [review]
All looking fantastic @mmioana ✨ 

## 1. Pagination doesn't reset back to 1

table no reset

## 2. Going from mobile to desktop closes the expanded contents

table mobile to tablet

## 3. Table dividers are visible on Safari

Screenshot 2024-11-01 at 00 54 03

## 4. Table item popover menu always visible

tablet not cut off

**Review by iamsamgibbs — APPROVED** [review]
Nice work on this!

No longer getting reset to page 1:

https://github.com/user-attachments/assets/a2fbed16-1dc5-48c5-9666-c363a7cdbf56

Rows no longer stay expanded:

https://github.com/user-attachments/assets/3013c8e8-3db4-45bd-823b-505e8b1770e2

Dividers are showing correctly:

Screenshot 2024-11-01 at 11 50 03

Popover menus now correctly show entire menu:
Screenshot 2024-11-01 at 11 52 43

I did notice this issue whilst testing (only on Safari):

Screenshot 2024-11-01 at 11 51 23

And this PR overlaps a tiny bit with the work done here: #3579 So maybe that issue should wait until this is merged, or could be fixed here.

Happy to approve this as is though as it resolves the issues you've outlined, and these two can be addressed in separate PRs.

**Pull request merged**